lib/Net/Async/Blockchain/BTC.pm

```diff
@@ -174,18 +175,20 @@ you received.
 async sub recursive_search {
     my ($self) = @_;
 
-    return undef unless $self->base_block_number;
+    return undef unless $self->block->number;
 
     my $current_block = await $self->rpc_client->get_last_block();
 
     return undef unless $current_block;
 
     while (1) {
-        last unless $current_block > $self->base_block_number;
-        my $block_hash = await $self->rpc_client->get_block_hash($self->base_block_number + 0);
+        last unless $current_block > $self->block->number;
+        my $block_hash = await $self->rpc_client->get_block_hash($self->block->number + 0);
```

Comment:
```suggestion
        my $block_hash = await $self->rpc_client->get_block_hash($self->block->number + 0);
         $self->source->emit($self->block);
```
IMHO, better to emit the block number before start processing it, in case the service has restarted again while pressing the block

Reply:
Is that why I'm calling up in ` $self->source->emit($self->block->up());` so if we can't reach this point the previous block will be processed again.

Reply:
but what about if the Redis value was empty when we start the service, it suppose read the block number from the DB
and if the DB value has changed after running the service (by receiving a new transaction from the new block)
and the recursive search block has not completed yet (the block number has not set in Redis)? IMHO, for more safety better to store before starting